### PR TITLE
Enable writing to rush build cache during Linux CI builds

### PR DESCRIPTION
**Pull request by ben-polinsky**
Due to ADO build-time vs runtime variable evaluation, the remote rush build cache was never written. This ensures the appropriate variable is set at runtime to enable write access for Linux `IndividualCI` and `Manual` builds.

### CODE REVIEW

File: common/scripts/set-rush-write-cache-variables.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ */
+
+const acceptableReasons = ["IndividualCI", "Manual"]
+let useRushWriteCache = 0
+
+if (process.env.AGENT_OS === 'Linux' && acceptableReasons.includes(process.env.BUILD_REASON))
+  useRushWriteCache = 1
+
+console.log(`setting RUSH_BUILD_CACHE_WRITE_ALLOWED to ${useRushWriteCache}`)
+console.log(`##vso[task.setvariable variable=RUSH_BUILD_CACHE_WRITE_ALLOWED;]${useRushWriteCache}`)
```
Comment (aruniverse) [inline common/scripts/set-rush-write-cache-variables.js:14]: is this supposed to be in a log? where does it actually set the env var?

**Review by aruniverse — APPROVED** [review]

File: common/scripts/set-rush-write-cache-variables.js (hunk 1 above)
Comment (ben-polinsky) [reply common/scripts/set-rush-write-cache-variables.js:14]: The `##vso...` line sets the variable - it's the equivalent of `echo` in `bash` or `Write-Host` in `powershell`:
https://github.com/iTwin/itwinjs-core/pull/4264

Comment (aruniverse) [reply common/scripts/set-rush-write-cache-variables.js:14]: TIL!

Comment (ben-polinsky) [reply common/scripts/set-rush-write-cache-variables.js:14]: Me too, actually. Thought it was restricted to bash/ps - but this makes sense.

**Comment by ben-polinsky** [comment]
@calebmshafer Last piece of the rush write cache, fyi, if you could review when you have a chance.

**Review by calebmshafer — APPROVED** [review]
LGTM

**Pull request merged**